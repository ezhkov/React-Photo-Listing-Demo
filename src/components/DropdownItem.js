import React from 'react';

const DropdownItem = ({onFilterClick, activeFilter, filterType, hasPadding}) => <div
    onClick={onFilterClick}
    className={`dropdown-item ${activeFilter === filterType ? 'active' : ''}`}
    style={(hasPadding) ? {paddingLeft: '34px'} : {}}
    >
    {filterType}
  {(activeFilter === filterType) && <i className="check">âœ“</i>}
  </div>

export default DropdownItem