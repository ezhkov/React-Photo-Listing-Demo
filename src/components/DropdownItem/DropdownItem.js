import React from 'react';
import PropTypes from 'prop-types';

const DropdownItem = ({
  onFilterClick,
  activeFilter,
  filterType,
  hasPadding,
}) => (
  <div
    role="presentation"
    onClick={onFilterClick}
    className={`dropdown-item ${activeFilter === filterType ? 'active' : ''}`}
    style={(hasPadding) ? { paddingLeft: '34px' } : {}}
  >
    {filterType}
    {(activeFilter === filterType) && <i className="check">âœ“</i>}
  </div>);

DropdownItem.propTypes = {
  onFilterClick: PropTypes.func,
  activeFilter: PropTypes.string,
  filterType: PropTypes.string,
  hasPadding: PropTypes.bool,
};
DropdownItem.defaultProps = {
  onFilterClick: PropTypes.func,
  activeFilter: '',
  filterType: '',
  hasPadding: false,
};


export default DropdownItem;
