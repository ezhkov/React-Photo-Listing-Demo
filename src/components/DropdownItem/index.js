import React from 'react';
import PropTypes from 'prop-types';

const DropdownItem = ({
  onFilterClick,
  activeFilter,
  filterType,
  hasPadding,
}) => (
  <div
    role="presentation"
    onClick={onFilterClick}
    className={`dropdown-item ${hasPadding ? 'pl-5' : ''} ${activeFilter === filterType ? 'active' : ''}`}
  >
    {filterType}
    {(activeFilter === filterType) && <i className="check">âœ“</i>}
  </div>);

DropdownItem.propTypes = {
  onFilterClick: PropTypes.func,
  activeFilter: PropTypes.string,
  filterType: PropTypes.string,
  hasPadding: PropTypes.bool,
};
DropdownItem.defaultProps = {
  onFilterClick: PropTypes.func,
  activeFilter: '',
  filterType: '',
  hasPadding: false,
};


export default DropdownItem;
