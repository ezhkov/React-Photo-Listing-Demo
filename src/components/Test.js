import React, { Component } from 'react';
import ClickOutside from 'react-click-outside';

class Test extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showDropdown: false,
    };
    this.toggle = this.toggle.bind(this);
    this.hide = this.hide.bind(this);
  }

  toggle() {
    const { showDropdown } = this.state;
    this.setState({ showDropdown: !showDropdown });
  }

  hide() {
    this.setState({ showDropdown: false });
  }

  render() {
    const { showDropdown } = this.state;
    return (
      <div>
        <div>
          <ClickOutside onClickOutside={this.hide}>
            <button type="button" onClick={this.toggle}>Button</button>
            { showDropdown && (
              <p>
                Im a menu or something that you want
                to hide when clicking outside.
              </p>)
            }
          </ClickOutside>
        </div>
        <div>
          <ClickOutside onClickOutside={this.hide}>
            <button type="button" onClick={this.toggle}>Button</button>
            { showDropdown && (
              <p>
                Im a menu or something that you want
                to hide when clicking outside.
              </p>)
            }
          </ClickOutside>
        </div>
      </div>
    );
  }
}

export default Test;
