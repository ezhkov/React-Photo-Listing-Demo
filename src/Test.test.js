import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Test from './components/Test';
import DropdownItem from './components/DropdownItem';

Enzyme.configure({ adapter: new Adapter() });


function setup() {
  const enzymeWrapper = shallow(<Test />);
  return {
    enzymeWrapper,
  };
}

describe('components', () => {
  describe('Test 1', () => {
    it('Test 2', () => {
      const { enzymeWrapper } = setup();

      expect(enzymeWrapper.find('div').hasClass('test')).toBe(true);

    });

    it('Test padding of div', () => {
      //const { enzymeWrapper } = setup();
      //const onClick = jest.fn(() => 1);
      const props = {
        onFilterClick: () => {},
        activeFilter: '',
        filterType: '',
        hasPadding: false,
      }
      const dropdownItem = shallow(<DropdownItem {...props} />)

      expect(dropdownItem.find('div').prop('role')).toEqual('presentation')
      expect(dropdownItem.find('div').prop('style')).toEqual({})
      
    });


    it('Test active of class of div', () => {
      const props1 = {
        onFilterClick: () => {},
        activeFilter: 'filter',
        filterType: 'filter',
        hasPadding: false,
      }
      const dropdownItem1 = shallow(<DropdownItem {...props1} />)

      expect(dropdownItem1.find('div').hasClass('active')).toEqual(true);
      

      const props2 = {
        onFilterClick: () => {},
        activeFilter: 'filter1',
        filterType: 'filter2',
        hasPadding: false,
      }
      const dropdownItem2 = shallow(<DropdownItem {...props2} />)

      expect(dropdownItem2.find('div').hasClass('active')).toEqual(false);
    });


    it('Test element of check', () => {
      const props1 = {
        onFilterClick: () => {},
        activeFilter: 'filter',
        filterType: 'filter',
        hasPadding: false,
      }
      const dropdownItem1 = shallow(<DropdownItem {...props1} />)

      expect(dropdownItem1.find('i').length).toEqual(1);
      

      const props2 = {
        onFilterClick: () => {},
        activeFilter: 'filter1',
        filterType: 'filter2',
        hasPadding: false,
      }
      const dropdownItem2 = shallow(<DropdownItem {...props2} />)

      expect(dropdownItem2.find('i').length).toEqual(0);
    });

    it('Test text of div element', () => {
      const props = {
        onFilterClick: () => {},
        activeFilter: 'filter',
        filterType: 'filter',
        hasPadding: false,
      }
      const dropdownItem = shallow(<DropdownItem {...props} />)

      expect(dropdownItem.find('div').text()).toEqual(props.filterType + 'âœ“')
    });

    it('Test method of div element', () => {

      function func() {
        const props = {
          onFilterClick: jest.fn(),
        }     
        const enzymeWrapper = shallow(<DropdownItem {...props} />)
        return {
          props,
          enzymeWrapper
        }
      }
      const { enzymeWrapper, props } = func()

      const div = enzymeWrapper.find('div')
      div.props().onClick()
      expect(props.onFilterClick).toBeCalled();
      expect(props.onFilterClick.mock.calls.length).toBe(1)
    });

    it('Test method of div element', () => {
      function funcTest() {
        const enzymeWrapper = shallow(<Test />);
        return {
          enzymeWrapper,
        };
      }
      const mock = jest.fn();
      mock.mockReturnValue(1);

      const { enzymeWrapper } = funcTest();
      const div = enzymeWrapper.find('div')
      
      expect(mock(div.prop.onClick)).toBe(1);

      //console.log(enzymeWrapper.debug());
      //expect(enzymeWrapper.find('div').hasClass('test')).toBe(true);

    });
  });
});