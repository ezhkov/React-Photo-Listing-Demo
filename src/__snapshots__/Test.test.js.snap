// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Test 1 simulates click events 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Test />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Text",
      "className": "test",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "Text",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Text",
        "className": "test",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "Text",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
